File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberService.java
package com.example.webmvc.favouriteNumber;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface FavouriteNumberService {
    FavouriteNumber saveNumber(FavouriteNumber number);

    List<FavouriteNumber> getPaginated(int results_per_page, int page);
    List<FavouriteNumber> getAll();

    Optional<FavouriteNumber> findByID(long id);

    long getNumberCount();

    void delete(FavouriteNumber number);

    FavouriteNumber getById(long id);
}

File: src/main/java/com/example/webmvc/favouriteNumber/FavouriteNumberService.kt
package com.example.webmvc.favouriteNumber

import org.springframework.stereotype.Service
import java.util.*

@Service
interface FavouriteNumberService {
    val all: List<FavouriteNumber>?
    val numberCount: Long

    fun saveNumber(number: FavouriteNumber): FavouriteNumber?
    fun getPaginated(results_per_page: Int, page: Int): List<FavouriteNumber>?
    fun findByID(id: Long): Optional<FavouriteNumber>?
    fun delete(number: FavouriteNumber?)
    fun getById(id: Long): FavouriteNumber?
}